contrib/babelfishpg_common/src/varbinary.c

```diff
@@ -732,12 +731,18 @@ varbinaryvarchar(PG_FUNCTION_ARGS)
 	VarChar    *result;
 	char 	   *encoded_result;
 	size_t		len = VARSIZE_ANY_EXHDR(source);
-	int32		typmod = PG_GETARG_INT32(1);
-	int32		maxlen = typmod - VARHDRSZ;
+	int32		typmod = -1;
+	int32		maxlen = -1;
```

Comment:
is this desired behavior to use typmod and maxlen as -1 by default? do we have test case which validates that case works as expected? have we found why tests were not failing with garbage typmod and maxlen?

Reply:
If we call this function without typmod (nargs = 1), According to the below condition, we need to call `encoding_conv_util` with the length of the source (len) as argument. Hence, -1 should work fine in this case. 

```
if (maxlen < 0 || len <= maxlen)
     encoded_result = encoding_conv_util(data, len, collInfo.enc, PG_UTF8, &encodedByteLen);
else
     encoded_result = encoding_conv_util(data, maxlen, collInfo.enc, PG_UTF8, &encodedByteLen);
```

The garbage values we were getting before were very large and hence, we will be always calling `encoding_conv_util` with the length of the source (len) as argument. We did not have any tests to test this scenario. Hence, the tests weren't failing before.

Reply:
if we cast 'abcdef' with varchar(2), it should be 'ab' but what will we get if we call function with 1 argument? we can add a test case in upgrade context where we will get incorrect result ('abcdef' instead of 'abc')

Reply:
I think do not have a test cases with input string length which is greater than these arbitrary large garbage values. But we should have test cases at least for varchar(max)